views/js/runner/plugins/navigation/next.js

```diff
@@ -96,41 +101,85 @@ define([
         init : function init(){
             var self = this;
             var testRunner = this.getTestRunner();
-
-            function doNext(context) {
-                if(context.isLast){
-                    self.trigger('end');
-                }
-                testRunner.next();
-            }
+            var testData = testRunner.getTestData();
+            var testConfig = testData.config || {};
+            var pluginShortcuts = (testConfig.shortcuts || {})[this.getName()] || {};
 
             //create the button (detached)
             this.$element = createElement(testRunner.getTestContext());
 
             //plugin behavior
-            this.$element.on('click', function(e){
-                var enable = _.bind(self.enable, self);
+            /**
+             * @param {Boolean} nextItemWarning - enable the display of a warning when going to the next item.
+             * Note: the actual display of the warning depends on other conditions (see nextWarningHelper)
+             */
+            function doNext(nextItemWarning) {
```

Comment:
You should keep the function declarations on top, mixing instructions and declaration can lead to bad reading.